### ISSUE TITLE
Spreading record in operation parameter cause issue in resolving path parameters

### ISSUE THREAD

**Issue by timotheeguerin** [microsoft/typespec#3051]
[Playground Link](https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwOwoKQHJvdXRlKCIvZm9vcy97Zm9vSWR9IikKb3AgdGVzdCgKICBAcGF0aCDFGjogc3RyaW5nLAoKICAvLyBjb21tZW50IGJlbG93IHRvIGZpeAogIC4uLlJlY29yZDzGLz4sCik6IHZvaWQ7Cg%3D%3D&e=%40typespec%2Fopenapi3&options=%7B%7D)

**Comment by qiaozha** [comment]
I wonder if the above typespec means we have an operation that have a body parameter which is a dictionary of string or it means we have additional arguments that takes string? Is it possible that there might be a scenario, service wants to have a open dictionary as query parameter? 
```
@query // this is not allowed now.
scalar bar extends string;

@route("/foos/{fooId}")
op test(
  @path fooId: string,
  ...Record<bar>,
): void;

```

**Comment by timotheeguerin** [comment]
Yeah this particular case means the input takes many parameters that all go in the body but we do have a separate issue on dictionary for queries
